### ISSUE TITLE
Display names for array entries in Inspector for JSON attributes

### ISSUE THREAD

**Issue by mvaligursky** [playcanvas/editor#210]
When an array is displayed, we see each entry called [0], [1] ...

<img width="416" alt="Screen Shot 2020-11-02 at 3 05 06 PM" src="https://user-images.githubusercontent.com/59932779/97883477-d3860e80-1d1c-11eb-80af-9cb5832da0d8.png">

Perhaps if element contains a string attribute, it could be used to enhance it.

For example a first attribute of type string should be used .. so if we have Name: john
it would show [0]: John

**Comment by willeastcott** [comment]
I don't think we can make this assumption. For example, check out the tween script and its first string attibute:

https://github.com/playcanvas/engine/blob/master/examples/assets/scripts/animation/tween.js#L13

It's the name of the event that triggers the tween. I think it'd be pretty weird to have that as the 'name' of the array element.

**Comment by mvaligursky** [comment]
If people know thata's the rule (like it is in some other engine), they would put the string they want as first.
But some more flexible solution we can do would work even better.

**Comment by yaustar** [comment]
Maybe we can an Editor only field above/below the array size that takes a string for the property in the object of each entry that we would like to display.
